### PR TITLE
bpo-46502: Remove "How do I tell incomplete input" from FAQ

### PR DESCRIPTION
**Pull request by mloskot**
Since,

- Py_CompileString no longer allows to distinguish "incomplete input"
  from "invalid input"
- there is no alternative solution available from the Python C API
  due to how the new parser works (rewritten in 3.9)
- the only supported way is to manually import the codeop module from C
  and use its API as IDLE does, and accept its own complications

it is desirable to remove this Q&A from the official FAQ.

------

See original report and related discusion at https://bugs.python.org/issue46502

https://bugs.python.org/issue46502

Automerge-Triggered-By: GH:pablogsal

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@mloskot

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/faq/extending.rst [hunk 1]
```diff
@@ -254,146 +254,6 @@ For Red Hat, install the python-devel RPM to get the necessary files.
 
 For Debian, run ``apt-get install python-dev``.
 
-
-How do I tell "incomplete input" from "invalid input"?
```
Comment (pablogsal) [inline Doc/faq/extending.rst:258]: Wait, we are removing a lot here. There is still the parts about `codeop` and `PyRun_InteractiveLoop` that are useful. We need to remove from this part forward:

>However sometimes you have to run the embedded Python interpreter in the same thread as your rest application and you can't allow the...

**Comment by pablogsal** [comment]
Since this is removing code, don't worry about the CLA, but is good if you want to contribute in the future.

File: Doc/faq/extending.rst (hunk 1 above)
Comment (mloskot) [reply Doc/faq/extending.rst:258]: @pablogsal Ha! Clearly, I've gone too eagerly too far. I've restored the relevant part. Thank you.

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
Thanks for the PR @mloskot !

**Comment by mloskot** [comment]
Thank you @pablogsal for backporting the changes.

**Pull request merged**